### ISSUE TITLE
0.17.0 reports Druid requires Java 8

### ISSUE THREAD

**Issue by ceastman-ibm** [apache/druid#9276]
Please provide a detailed title (e.g. "Broker crashes when using TopN query with Bound filter" instead of just "Broker crashes").

### Affected Version

0.17.0

### Description

Please include as much detailed information about the problem as possible.
- Cluster size
- Configurations in use
- Steps to reproduce the problem
- The error message or stack traces encountered. Providing more context, such as nearby log messages or even entire logs, can be helpful.
- Any debugging that you have already done

Carls-MBP:druid carleastman$ kubectl logs druid-789d7d4496-c2g5f
sh: 1: source: not found
Druid requires Java 8. Make sure that "java" is installed and on your PATH.

If you believe this check is in error, you can skip it using an
environment variable:

  export DRUID_SKIP_JAVA_CHECK=1

Otherwise, install Java 8 and try again.

This script searches for Java 8 in 3 locations in the following
order
  * DRUID_JAVA_HOME
  * JAVA_HOME
  * java (installed on PATH)
Carls-MBP:druid carleastman$ kubectl exec -it druid-7649d9d7dd-9hbjg bash

root@druid-7649d9d7dd-9hbjg:/# java -version
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (build 1.8.0_242-8u242-b08-0ubuntu3~18.04-b08)
OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)

**Comment by suneet-s** [comment]
@ceastman-ibm Is JAVA_HOME set on your machine? If so druid will try to use that instead of java installed on the path

**Comment by ceastman-ibm** [comment]
@suneet-s JAVA_HOME isn't set.

**Comment by suneet-s** [comment]
@ceastman-ibm there is a utility script in the bin folder called java_util that druid runs to find java. You can run that to see where druid thinks java is

`source java-util && get_java_bin_dir`

**Comment by ceastman-ibm** [comment]
root@druid-67b97f4d87-bl9f2:/druid/bin# source java-util && get_java_bin_dir
/usr/bin

**Comment by suneet-s** [comment]
@ceastman-ibm I just noticed this in the logs

`sh: 1: source: not found`

Looks like this is a bug in the startup script - https://stackoverflow.com/questions/13702425/source-command-not-found-in-sh-shell/13702876

To get you unblocked, I'd recommend you skip the java check in the startup script by setting `export DRUID_SKIP_JAVA_CHECK=1` till this issue is fixed.

Thanks for reporting!

**Comment by ceastman-ibm** [comment]
@suneet-s yep i found the workaround and implemented that, so i am unblocked now. thank you. i also saw the java 11 was going to be supported shortly.

**Comment by jbs123** [comment]
I am getting the same issues then it fails to start. 
Fresh install using Ubuntu 18 with open-jdk 8. 
$ echo $JAVA_HOME
/usr/lib/jvm/zulu-8-amd64/jre/bin/
$ which java
/usr/bin/java
$ java -version
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (Zulu 8.44.0.11-CA-linux64) (build 1.8.0_242-b20)
OpenJDK 64-Bit Server VM (Zulu 8.44.0.11-CA-linux64) (build 25.242-b20, mixed mode)

**Comment by suneet-s** [comment]
If you run a zsh shell - this is broken as well 😭

**Comment by mholleville** [comment]
Hello ! This issue is closed, however I still have the same error message as @ceastman-ibm. Do you have more info?

I have the error only when I try to launch druid components separately
